osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -151,6 +151,41 @@ public async Task AddItem(MultiplayerPlaylistItem item, MultiplayerRoomUser user
             }
         }
 
+        /// <summary>
+        /// Removes a playlist item from the room's queue.
+        /// </summary>
+        /// <param name="playlistItemId">The item to remove.</param>
+        /// <param name="user">The user removing the item.</param>
+        public async Task RemoveItem(long playlistItemId, MultiplayerRoomUser user)
+        {
+            if (dbFactory == null) throw new InvalidOperationException($"Call {nameof(Initialise)} first.");
+
+            if (room.Settings.QueueMode == QueueMode.HostOnly)
+                throw new InvalidStateException("Items cannot be removed in host-only mode.");
```

Comment:
Yeah, I'm not sure about this either. With the way how I understand queue mode to work (in that items from previous queue mode stay there after change), you could have a situation wherein a room was in round robin and there were several items enqueued by users, and then if the host changes to host-only, nobody can remove any of those items anymore, host included.

A simple compromise may be to drop this check altogether and rely on the "cannot delete current item" check for sanity. I think that should be enough to ensure users don't get themselves into a broken state? That's also just my opinion, probably a good idea to wait for another.

Reply:
> drop this check altogether

I agree with that also.

> Also not sure that the client should be authoritative with the display of the delete button

Is there a reason why the server should be authoritative on everything? How far do you go with this - should there also be a flag telling the client whether they can kick participants rather than the host check? Should there be a flag telling the client whether they can add items to the queue rather than the queue mode + host check? Should there be a flag telling the client whether they can ready up and/or start the match based on the conditions currently in `MultiplayerReadyButton`?  
It feels like unnecessary complexity to me.

And to illustrate the complexity, if it were to happen, how would you envision such a flag being passed to clients? Would there be a dictionary in `MultiplayerPlaylistItem` mapping each user id in the room to whether they can delete the item? Or would there be a flag in `MultiplayerSettings` telling all clients whether items _may be possible to be_ deleted, but then the clients would still have to do their own `OwnerID` check?

Reply:
The answer is "no" to all the above. The reason for it here was that the conditionals here are (were?) quite complicated.

The objective was to simplify things, not complicate. The flag would be sent as part of the `MultiplayerPlaylistItem`, not externally.

Reply:
If the objective were to simplify, then this is not it. Instead of 4 lines in one class, one of which was redundant, and which is reduced to 2 in my upcoming changes, you'd have multiple lines across `MultiplayerRoom`, `MultiplayerClient`, `PlaylistItem`, `TestMultiplayerClient`, and `DrawableRoomPlaylistItem` just to get such a flag to work.

So I disagree with that path.